### DISCUSSION TITLE
Enabling Read-Write functions on Vercel Deployment

### DISCUSSION THREAD

**Discussion by TASLIM-ENIOLLA** [vercel/next.js#46821] [Help]
### Summary

Hello Devs. I recently created an app and the app has api routes. In these api routes, the 'fs' module was used to read-write data. The reading of data goes well but when it comes to writing data, the log shows an error that says only reading is allowed. How do I configure my vercel deploymant to be read-writeable?

**Comment by icyJoseph** [comment]
Hi, 

You can't really write, anymore than to `tmp`. https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage

Fundamentally, the issue is that these are all serverless deployments, meaning that there's no actual server running all of the time, nor a dedicated computer associated with your deployment.

Rather, given your code deploy, as the requests come in, a new instance of your code is spun-up, and after a given time it is shut down, and with it goes all of the information generated during that time.

- https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage
- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002

Take this extract from here: https://vercel.com/guides/is-sqlite-supported-in-vercel#local-storage-and-serverless,

> SQLite needs a local file system on the server to store the data permanently when write requests are made. In a serverless environment, this central single permanent storage is not available because storage is ephemeral with serverless functions. As a function receives more concurrent traffic, the serverless environment will create new instances of the function and each instance will not be able to share the same storage.
